matchers/compliance/compliance_test.go

```diff
@@ -0,0 +1,376 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build compliance
```

Comment:
We should remove this tag - we should always run these tests.

Reply:
The issue is some of them fail on purpose to highlight the breaking changes between old and new parsers, so will cause CI to fail. What should we do about those?

Reply:
Let's skip the cases that fail with a comment.